org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/Conversions.java

```diff
@@ -61,139 +189,305 @@ public static Object doWrapArray(Object object) {
 
 		if (!arrayClass.isArray()) {
 			// Can't wrap something that isn't an array.
+//			throw new IllegalArgumentException("Can't wrap something that isn't an array.");
 			return object;
```

Comment:
is it ok to return non wrapped object?
if not we change return type to Iterable<?>

Reply:
This would change the contract of the method. I don't think that's possible.